integreat_cms/cms/utils/internal_link_utils.py

```diff
@@ -51,14 +60,14 @@ def update_link_language(
         # Always use the full url, even if the url was previously a short url
         fixed_link = target_translation.full_url
 
-        # Update the title if it was previously the translation title or the url
-        if link_text:
-            if source_translation.title.lower() == link_text.strip().lower():
-                link_text = target_translation.title
-            elif current_link.strip() == link_text.strip():
-                link_text = fixed_link
+        # Update the title if it was previously the url, otherwise use the new title
+        link_html = None
+        if len(link) == 0 and link.text and current_url.strip() == link.text.strip():
```

Comment:
What does the len(link) check for? Isn't the link an Element?

Reply:
`# Update the title if it was previously the url `

I haven't looked into it deeply, but it seems to work for pages, but not for events or locations 🤔